pkg/app/piped/cloudprovider/kubernetes/kubernetes.go

```diff
@@ -244,7 +244,16 @@ func (p *provider) Delete(ctx context.Context, k ResourceKey) (err error) {
 		return p.initErr
 	}
 
-	return p.kubectl.Delete(ctx, p.input.Namespace, k)
+	return p.kubectl.Delete(ctx, p.getNamespaceForRun(k), k)
+}
+
+// getNamespaceForRun returns namespace used on kubectl apply/delete commands
+// priority: config.KubernetesDeploymentInput > kubernetes.ResourceKey
+func (p *provider) getNamespaceForRun(k ResourceKey) string {
+	if p.input.Namespace != "" {
+		return p.input.Namespace
+	}
+	return k.Namespace
```

Comment:
We could get the namespace from ResourceKey ( from manifest files ) or from KubernetesDeploymentInput ( which could be empty on unset currently ). IMO, namespace value from KubernetesDeploymentInput should have a higher priority than the value from ResourceKey, in case of mismatch, the current `not found resources to delete` error will be raised.
How do you think?

Reply:
I think the one in the user-defined manifest should be more prioritized than `pipe.yaml` when applying.

Reply:
To be simple, the piped's role is basically just to perform `kubectl apply -f manifest.yaml`.

Reply:
Hmm... but on the second thought, the case that user clearly defines its namespace in the `.pipe.yaml`, I'm begging to feel it's better to override the base manifest.

Reply:
One vote from me for taking the namespace value from the Input with higher priority.

Reply:
> Hmm... but on the second thought, the case that user clearly defines its namespace in the `.pipe.yaml`, I'm begging to feel it's better to override the base manifest.

yep, I feel the same either. Since the manifest should be treated as a base, and config on `.pipe.yaml` lay on that base and who defined the config should be aware of where should the deployment they set will be applied to.